fixture-monkey-kotlin/src/main/kotlin/com/navercorp/fixturemonkey/kotlin/introspector/KotlinDurationIntrospector.kt

```
@@ -0,0 +1,58 @@
+ */
+package com.navercorp.fixturemonkey.kotlin.introspector
+
+import com.navercorp.fixturemonkey.api.arbitrary.CombinableArbitrary
+import com.navercorp.fixturemonkey.api.generator.ArbitraryGeneratorContext
+import com.navercorp.fixturemonkey.api.introspector.ArbitraryIntrospector
+import com.navercorp.fixturemonkey.api.introspector.ArbitraryIntrospectorResult
+import com.navercorp.fixturemonkey.api.matcher.Matcher
+import com.navercorp.fixturemonkey.api.property.Property
+import com.navercorp.fixturemonkey.kotlin.matcher.Matchers.DURATION_TYPE_MATCHER
+import kotlin.reflect.full.primaryConstructor
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+class KotlinDurationIntrospector : ArbitraryIntrospector, Matcher {
+    override fun match(property: Property) = DURATION_TYPE_MATCHER.match(property)
+
+    override fun introspect(context: ArbitraryGeneratorContext): ArbitraryIntrospectorResult {
+        val kClass = Duration::class
+        val primaryConstructor = kClass.primaryConstructor
+
+        require(match(context.resolvedProperty)) { "Given type is not Duration type: " + context.resolvedType }
+        require(primaryConstructor != null) { "Duration class must have primary constructor" }
+        require(primaryConstructor.parameters.size == 1) { "Duration class must have only one parameter" }
+
+        val parameterName = primaryConstructor.parameters[0].name
+
+        return ArbitraryIntrospectorResult(
+            CombinableArbitrary.objectBuilder()
+                .properties(context.combinableArbitrariesByArbitraryProperty)
+                .build {
+                    val parameterValue = it.mapKeys { map -> map.key.objectProperty.property.name }[parameterName]
+                    if (parameterValue is Long) {
+                        parameterValue.toDuration(DurationUnit.values().random())
+                    } else {
+                        Duration.ZERO
+                    }
```

It would be better to throw an exception if it is not `Long` type.

---

Changed to throw a cast exception if parameterValue is not Long.

```kotlin
...
val parameterValue = it
    .mapKeys { map -> map.key.objectProperty.property.name }[parameterName] as Long
parameterValue.toDuration(DurationUnit.values().random())
```